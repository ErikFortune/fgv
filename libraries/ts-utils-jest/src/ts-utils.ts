export * from '@fgv/ts-utils';
