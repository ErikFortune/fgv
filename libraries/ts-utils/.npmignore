# Agent-specific files
CLAUDE.md
.agents/
.claude/

# IDE and system files
.vscode/
.DS_Store

# Test and coverage files  
coverage/
*.test.ts
*.test.js
*.test.d.ts
*.test.*.map
src/test/
lib/test/
lib/**/*.test.*
lib-commonjs/test/
lib-commonjs/**/*.test.*

# Development files
*.log
.eslintrc*
jest.config.*
tsconfig.json
*.map

# Source files (if publishing only built output)
# Uncomment if you only want to publish dist/lib
src/

# Rush/Heft temporary files
.heft/
.rush/
rush-logs/
temp/

# API Extractor files (development artifacts)
etc/

# Build artifacts that shouldn't be published
dist/
lib-commonjs/

# Documentation source
docs/
*.md
!README.md
!LICENSE.md