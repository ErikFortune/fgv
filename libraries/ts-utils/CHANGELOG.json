{
  "name": "@fgv/ts-utils",
  "entries": [
    {
      "version": "4.2.0",
      "tag": "@fgv/ts-utils_v4.2.0",
      "date": "Mon, 20 Jan 2025 09:46:53 GMT",
      "comments": {
        "none": [
          {
            "comment": "add collections packlet and ResultMap class"
          },
          {
            "comment": "add withFormattedError on validator and converter"
          },
          {
            "comment": "clean up type inconsistencies, relax some constraints"
          }
        ]
      }
    },
    {
      "version": "4.1.0",
      "tag": "@fgv/ts-utils_v4.1.0",
      "date": "Thu, 09 Jan 2025 05:33:39 GMT",
      "comments": {
        "none": [
          {
            "comment": "update dependencies"
          }
        ]
      }
    },
    {
      "version": "4.0.2",
      "tag": "@fgv/ts-utils_v4.0.2",
      "date": "Tue, 14 May 2024 14:45:53 GMT",
      "comments": {}
    },
    {
      "version": "4.0.1",
      "tag": "@fgv/ts-utils_v4.0.1",
      "date": "Tue, 14 May 2024 05:02:20 GMT",
      "comments": {
        "none": [
          {
            "comment": "publish"
          },
          {
            "comment": "import TextEncoder to avoid globals race"
          }
        ]
      }
    },
    {
      "version": "4.0.0",
      "tag": "@fgv/ts-utils_v4.0.0",
      "date": "Tue, 14 May 2024 03:09:27 GMT",
      "comments": {
        "none": [
          {
            "comment": "rework validation/converter compatibility a bit"
          },
          {
            "comment": "QoL improvement to simplify error aggregation"
          },
          {
            "comment": "factor optional converters out to ts-extras package to eliminate large dependencies"
          },
          {
            "comment": "update generated api docs"
          },
          {
            "comment": "add convalidate helper"
          },
          {
            "comment": "add defaulting converter"
          },
          {
            "comment": "extend conversion to accept validators"
          },
          {
            "comment": "export StringConverter at top level"
          }
        ]
      }
    },
    {
      "version": "3.0.0",
      "tag": "@fgv/ts-utils_v3.0.0",
      "date": "Mon, 22 Jan 2024 07:00:18 GMT",
      "comments": {
        "none": [
          {
            "comment": "gitignore tweaks"
          },
          {
            "comment": "refactor hash implementation"
          },
          {
            "comment": "bump versions"
          },
          {
            "comment": "refactor and cleanup"
          },
          {
            "comment": "build changes"
          },
          {
            "comment": "Factor out extras, enable shaking"
          },
          {
            "comment": "bump dependencies"
          },
          {
            "comment": "version upgrades"
          },
          {
            "comment": "relax mapToRecord from Map to ReadonlyMap"
          },
          {
            "comment": "bump version"
          },
          {
            "comment": "add oneOf validator"
          }
        ]
      }
    },
    {
      "version": "2.0.0",
      "tag": "@fgv/ts-utils_v2.0.0",
      "date": "Mon, 10 Apr 2023 03:33:11 GMT",
      "comments": {
        "major": [
          {
            "comment": "refactor and cleanup"
          },
          {
            "comment": "reorganization and cleanup"
          },
          {
            "comment": "more refactoring"
          },
          {
            "comment": "refactor"
          }
        ]
      }
    }
  ]
}
