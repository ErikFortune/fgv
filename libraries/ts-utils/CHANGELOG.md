# Change Log - @fgv/ts-utils

This log was last generated on Thu, 30 Jan 2025 00:35:17 GMT and should not be manually modified.

## 4.3.0
Thu, 30 Jan 2025 00:35:17 GMT

### Updates

- add collector, convertingResultMap, convertingCollector

## 4.2.2
Thu, 23 Jan 2025 06:19:32 GMT

### Updates

- QoL improvements to Success/Failure

## 4.2.1
Tue, 21 Jan 2025 04:19:21 GMT

### Updates

- allow factories in getOrAdd

## 4.2.0
Mon, 20 Jan 2025 09:46:53 GMT

### Updates

- add collections packlet and ResultMap class
- add withFormattedError on validator and converter
- clean up type inconsistencies, relax some constraints

## 4.1.0
Thu, 09 Jan 2025 05:33:39 GMT

### Updates

- update dependencies

## 4.0.2
Tue, 14 May 2024 14:45:53 GMT

_Version update only_

## 4.0.1
Tue, 14 May 2024 05:02:20 GMT

### Updates

- publish
- import TextEncoder to avoid globals race

## 4.0.0
Tue, 14 May 2024 03:09:27 GMT

### Updates

- rework validation/converter compatibility a bit
- QoL improvement to simplify error aggregation
- factor optional converters out to ts-extras package to eliminate large dependencies
- update generated api docs
- add convalidate helper
- add defaulting converter
- extend conversion to accept validators
- export StringConverter at top level

## 3.0.0
Mon, 22 Jan 2024 07:00:18 GMT

### Updates

- gitignore tweaks
- refactor hash implementation
- bump versions
- refactor and cleanup
- build changes
- Factor out extras, enable shaking
- bump dependencies
- version upgrades
- relax mapToRecord from Map to ReadonlyMap
- bump version
- add oneOf validator

## 2.0.0
Mon, 10 Apr 2023 03:33:11 GMT

### Breaking changes

- refactor and cleanup
- reorganization and cleanup
- more refactoring
- refactor

