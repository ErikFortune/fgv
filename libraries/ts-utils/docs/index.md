<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md)

## API Reference

## Packages

<table><thead><tr><th>

Package


</th><th>

Description


</th></tr></thead>
<tbody><tr><td>

[@fgv/ts-utils](./ts-utils.md)


</td><td>


</td></tr>
</tbody></table>
