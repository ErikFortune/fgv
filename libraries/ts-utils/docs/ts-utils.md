<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@fgv/ts-utils](./ts-utils.md)

## ts-utils package

## Classes

|  Class | Description |
|  --- | --- |
|  [DetailedFailure](./ts-utils.detailedfailure.md) | A [DetailedFailure](./ts-utils.detailedfailure.md) extends [Failure](./ts-utils.failure.md) to report optional failure details in addition to the error message. |
|  [DetailedSuccess](./ts-utils.detailedsuccess.md) | A [DetailedSuccess](./ts-utils.detailedsuccess.md) extends [Success](./ts-utils.success.md) to report optional success details in addition to the error message. |
|  [Failure](./ts-utils.failure.md) | Reports a failed [result](./ts-utils.iresult.md) from some operation, with an error message. |
|  [Normalizer](./ts-utils.normalizer.md) | Normalizes an arbitrary JSON object |
|  [ObjectConverter](./ts-utils.objectconverter.md) | A [Converter](./ts-utils.converter.md) which converts an object of type <code>&lt;T&gt;</code> without changing shape, given a [FieldConverters&lt;T&gt;](./ts-utils.conversion.fieldconverters.md) for the fields in the object. |
|  [StringConverter](./ts-utils.stringconverter.md) | The [StringConverter](./ts-utils.conversion.stringconverter.md) class extends [BaseConverter](./ts-utils.conversion.baseconverter.md) to provide string-specific helper methods. |
|  [Success](./ts-utils.success.md) | Reports a successful [result](./ts-utils.iresult.md) from some operation and the corresponding value. |

## Functions

|  Function | Description |
|  --- | --- |
|  [allSucceed(results, successValue)](./ts-utils.allsucceed.md) | Determines if an iterable collection of [Result&lt;T&gt;](./ts-utils.result.md) were all successful. |
|  [captureResult(func)](./ts-utils.captureresult.md) | Wraps a function which might throw to convert exception results to [Failure](./ts-utils.failure.md)<!-- -->. |
|  [fail\_2(message)](./ts-utils.fail_2.md) | Returns [Failure&lt;T&gt;](./ts-utils.failure.md) with the supplied error message. |
|  [failWithDetail(message, detail)](./ts-utils.failwithdetail.md) | Returns [DetailedFailure&lt;T, TD&gt;](./ts-utils.detailedfailure.md) with a supplied error message and detail. |
|  [getTypeOfProperty(key, item)](./ts-utils.gettypeofproperty.md) | Gets the type of a property specified by key from an arbitrary object. |
|  [getValueOfPropertyOrDefault(key, item, defaultValue)](./ts-utils.getvalueofpropertyordefault.md) | Gets the value of a property specified by key from an arbitrary object, or a default value if the property does not exist. |
|  [isKeyOf(key, item)](./ts-utils.iskeyof.md) | Helper type-guard function to report whether a specified key is present in a supplied object. |
|  [mapDetailedResults(results, ignore)](./ts-utils.mapdetailedresults.md) | Aggregates successful results from a collection of [DetailedResult&lt;T, TD&gt;](./ts-utils.detailedresult.md)<!-- -->, optionally ignoring certain error details. |
|  [mapFailures(results)](./ts-utils.mapfailures.md) | Aggregates error messages from a collection of [Result&lt;T&gt;](./ts-utils.result.md)<!-- -->. |
|  [mapResults(results)](./ts-utils.mapresults.md) | Aggregates successful result values from a collection of [Result&lt;T&gt;](./ts-utils.result.md)<!-- -->. |
|  [mapSuccess(results)](./ts-utils.mapsuccess.md) | Aggregates successful results from a a collection of [Result&lt;T&gt;](./ts-utils.result.md)<!-- -->. |
|  [mapToRecord(src, factory)](./ts-utils.maptorecord.md) | Applies a factory method to convert a <code>ReadonlyMap&lt;TK, TS&gt;</code> into a <code>Record&lt;TK, TD&gt;</code>. |
|  [omit(from, exclude)](./ts-utils.omit.md) | Simple implicit omit function, which picks all of the properties from a supplied object except those specified for exclusion. |
|  [optionalMapToPossiblyEmptyRecord(src, factory)](./ts-utils.optionalmaptopossiblyemptyrecord.md) | Applies a factory method to convert an optional <code>ReadonlyMap&lt;string, TS&gt;</code> into a <code>Record&lt;string, TD&gt;</code> |
|  [optionalMapToRecord(src, factory)](./ts-utils.optionalmaptorecord.md) | Applies a factory method to convert an optional <code>ReadonlyMap&lt;string, TS&gt;</code> into a <code>Record&lt;string, TD&gt;</code> or <code>undefined</code>. |
|  [optionalRecordToMap(src, factory)](./ts-utils.optionalrecordtomap.md) | Applies a factory method to convert an optional <code>Record&lt;TK, TS&gt;</code> into a <code>Map&lt;TK, TD&gt;</code>, or <code>undefined</code>. |
|  [optionalRecordToPossiblyEmptyMap(src, factory)](./ts-utils.optionalrecordtopossiblyemptymap.md) | Applies a factory method to convert an optional <code>Record&lt;TK, TS&gt;</code> into a <code>Map&lt;TK, TD&gt;</code> |
|  [pick(from, include)](./ts-utils.pick.md) | Simple implicit pick function, which picks a set of properties from a supplied object. Ignores picked properties that do not exist regardless of type signature. |
|  [populateObject(initializers, options)](./ts-utils.populateobject.md) | Populates an an object based on a prototype full of field initializers that return [Result&lt;T\[key\]&gt;](./ts-utils.result.md)<!-- -->. Returns [Success](./ts-utils.success.md) with the populated object if all initializers succeed, or [Failure](./ts-utils.failure.md) with a concatenated list of all error messages. |
|  [populateObject(initializers, order)](./ts-utils.populateobject_1.md) | Populates an an object based on a prototype full of field initializers that return [Result&lt;T\[key\]&gt;](./ts-utils.result.md)<!-- -->. Returns [Success](./ts-utils.success.md) with the populated object if all initializers succeed, or [Failure](./ts-utils.failure.md) with a concatenated list of all error messages. |
|  [propagateWithDetail(result, detail, successDetail)](./ts-utils.propagatewithdetail.md) | Propagates a [Success](./ts-utils.success.md) or [Failure](./ts-utils.failure.md) [Result](./ts-utils.result.md)<!-- -->, adding supplied event details as appropriate. |
|  [recordToMap(src, factory)](./ts-utils.recordtomap.md) | Applies a factory method to convert a <code>Record&lt;TK, TS&gt;</code> into a <code>Map&lt;TK, TD&gt;</code>. |
|  [succeed(value)](./ts-utils.succeed.md) | Returns [Success&lt;T&gt;](./ts-utils.success.md) with the supplied result value. |
|  [succeedWithDetail(value, detail)](./ts-utils.succeedwithdetail.md) | Returns [DetailedSuccess&lt;T, TD&gt;](./ts-utils.detailedsuccess.md) with a supplied value and optional detail. |

## Interfaces

|  Interface | Description |
|  --- | --- |
|  [Converter](./ts-utils.converter.md) | Generic converter to convert unknown to a templated type <code>&lt;T&gt;</code>, using intrinsic rules or as modified by an optional conversion context of optional templated type <code>&lt;TC&gt;</code> (default <code>undefined</code>). |
|  [IResult](./ts-utils.iresult.md) | Represents the result of some operation of sequence of operations. |
|  [IResultLogger](./ts-utils.iresultlogger.md) | Simple logger interface used by [IResult.orThrow()](./ts-utils.iresult.orthrow.md)<!-- -->. |
|  [PopulateObjectOptions](./ts-utils.populateobjectoptions.md) | Options for the [populateObject()](./ts-utils.populateobject.md) function. |
|  [Validator](./ts-utils.validator.md) | In-place validation that a supplied unknown matches some required characteristics (type, values, etc). |

## Namespaces

|  Namespace | Description |
|  --- | --- |
|  [Conversion](./ts-utils.conversion.md) |  |
|  [Converters](./ts-utils.converters.md) |  |
|  [Hash](./ts-utils.hash.md) |  |
|  [Logging](./ts-utils.logging.md) |  |
|  [Validation](./ts-utils.validation.md) |  |
|  [Validators](./ts-utils.validators.md) |  |

## Type Aliases

|  Type Alias | Description |
|  --- | --- |
|  [Brand](./ts-utils.brand.md) | Helper type to brand a simple type to prevent inappropriate use |
|  [DetailedFailureContinuation](./ts-utils.detailedfailurecontinuation.md) | Callback to be called when a [DetailedResult](./ts-utils.detailedresult.md) encounters a failure. |
|  [DetailedResult](./ts-utils.detailedresult.md) | **_(BETA)_** Type inference to determine the result type <code>T</code> of a [DetailedResult&lt;T, TD&gt;](./ts-utils.detailedresult.md)<!-- -->. |
|  [DetailedSuccessContinuation](./ts-utils.detailedsuccesscontinuation.md) | Callback to be called when a [DetailedResult](./ts-utils.detailedresult.md) encounters success. |
|  [FailureContinuation](./ts-utils.failurecontinuation.md) | Continuation callback to be called in the event that an [Result](./ts-utils.result.md) fails. |
|  [FieldInitializers](./ts-utils.fieldinitializers.md) | String-keyed record of initialization functions to be passed to [populateObject()](./ts-utils.populateobject.md) or [populateObject()](./ts-utils.populateobject_1.md)<!-- -->. |
|  [Result](./ts-utils.result.md) | Represents the [result](./ts-utils.iresult.md) of some operation or sequence of operations. |
|  [ResultDetailType](./ts-utils.resultdetailtype.md) | **_(BETA)_** Type inference to determine the detail type <code>TD</code> of a [DetailedResult&lt;T, TD&gt;](./ts-utils.detailedresult.md)<!-- -->. |
|  [ResultValueType](./ts-utils.resultvaluetype.md) | **_(BETA)_** Type inference to determine the result type of an [Result](./ts-utils.result.md)<!-- -->. |
|  [SuccessContinuation](./ts-utils.successcontinuation.md) | Continuation callback to be called in the event that an [Result](./ts-utils.result.md) is successful. |

