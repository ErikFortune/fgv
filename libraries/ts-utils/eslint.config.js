// ESLint 9 flat config for ts-utils
const nodeProfile = require('@rushstack/eslint-config/flat/profile/node');
const packletsPlugin = require('@rushstack/eslint-config/flat/mixins/packlets');
const tsdocPlugin = require('@rushstack/eslint-config/flat/mixins/tsdoc');

module.exports = [
  ...nodeProfile,
  packletsPlugin,
  ...tsdocPlugin,
  {
    // Override specific rules if needed
    rules: {
      '@rushstack/packlets/mechanics': 'warn'
    }
  }
];
