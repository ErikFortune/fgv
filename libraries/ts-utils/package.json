{
  "name": "@fgv/ts-utils",
  "version": "3.0.1-alpha.3",
  "description": "Assorted Typescript Utilities",
  "main": "lib/index.js",
  "types": "dist/ts-utils.d.ts",
  "scripts": {
    "build": "heft test --clean",
    "clean": "heft clean",
    "test": "heft test",
    "build-docs": "api-documenter markdown --input-folder ./temp --output-folder docs",
    "build-all": "rushx build; rushx build-docs",
    "test-handles": "jest --runInBand --detectOpenHandles",
    "clean-jest": "jest --clear-cache",
    "coverage": "jest --coverage",
    "lint": "eslint src --ext .ts",
    "fixlint": "eslint src --ext .ts --fix"
  },
  "sideEffects": false,
  "keywords": [
    "typescript",
    "json"
  ],
  "author": "Erik Fortune",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/ErikFortune/fgv/issues"
  },
  "homepage": "https://github.com/ErikFortune/fgv/tree/main/libraries/ts-utils#readme",
  "devDependencies": {
    "@jest/expect-utils": "^29.7.0",
    "@microsoft/api-documenter": "^7.23.20",
    "@microsoft/api-extractor": "^7.39.4",
    "@types/jest": "^29.5.12",
    "@types/luxon": "^3.4.2",
    "@types/mustache": "^4.2.5",
    "@types/node": "^20.11.16",
    "@typescript-eslint/eslint-plugin": "^6.20.0",
    "@typescript-eslint/parser": "^6.20.0",
    "eslint": "^8.56.0",
    "eslint-config-standard": "^17.1.0",
    "eslint-plugin-import": "^2.29.1",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-promise": "^6.1.1",
    "jest": "^29.7.0",
    "jest-extended": "^4.0.2",
    "jest-matcher-utils": "^29.7.0",
    "rimraf": "^5.0.5",
    "ts-jest": "^29.1.2",
    "ts-node": "^10.9.2",
    "typescript": "^5.3.3",
    "eslint-plugin-n": "^16.6.2",
    "jest-snapshot": "~29.7.0",
    "@rushstack/heft": "~0.64.3",
    "@rushstack/heft-node-rig": "~2.4.5",
    "@rushstack/eslint-config": "~3.6.2",
    "@types/heft-jest": "1.0.6",
    "@rushstack/heft-jest-plugin": "~0.11.3",
    "eslint-plugin-tsdoc": "~0.2.17"
  },
  "dependencies": {
    "luxon": "^3.4.4",
    "mustache": "^4.2.0",
    "crc": "~4.3.2"
  }
}
