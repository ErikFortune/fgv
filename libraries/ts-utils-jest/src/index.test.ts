import './index';

import { fail, failWithDetail, succeed, succeedWithDetail } from '@fgv/ts-utils';

describe('verify matchers accessibility', () => {
  test('toFail exists and works', () => {
    expect(fail('oops')).toFail();
    expect(succeed('hi')).not.toFail();
  });

  test('toFailWith exists and works', () => {
    expect(fail('oops')).toFailWith('oops');
  });

  test('toFailWithDetail exists and works', () => {
    expect(failWithDetail<string, string>('oops', 'detail')).toFailWithDetail('oops', 'detail');
  });

  test('toSucceed exists and works', () => {
    expect(succeed('howdy')).toSucceed();
  });

  test('toSucceedWith exists and works', () => {
    expect(succeed('howdy')).toSucceedWith('howdy');
  });

  test('toSucceedWithDetail exists and works', () => {
    expect(succeedWithDetail('howdy', 'detail')).toSucceedWithDetail('howdy', 'detail');
  });

  test('toSucceedAndSatisfy exists and works', () => {
    expect(succeed('howdy')).toSucceedAndSatisfy((v: string) => {
      expect(v).toBe('howdy');
      return true;
    });
  });

  test('toSucceedAndMatchSnapshot exists and works', () => {
    expect(
      succeed({
        field: 'field',
        child: {
          values: [{ number: 1 }, { number: 2 }]
        }
      })
    ).toSucceedAndMatchSnapshot();
  });

  test('toSucceedAndMatchInlineSnapshot exists and works', () => {
    expect(
      succeed({
        field: 'field',
        child: {
          values: [{ number: 1 }, { number: 2 }]
        }
      })
    ).toSucceedAndMatchInlineSnapshot(`
            {
              "child": {
                "values": [
                  {
                    "number": 1,
                  },
                  {
                    "number": 2,
                  },
                ],
              },
              "field": "field",
            }
        `);
  });

  test('toFailTest exists and works', () => {
    expect(() => {
      expect(true).toBe(false);
    }).toFailTest();
  });

  test('toFailTestAndMatchSnapshot exists and works', () => {
    expect(() => {
      expect(true).toBe(false);
    }).toFailTestAndMatchSnapshot();
  });

  test('toFailTestWith exists and works', () => {
    expect(() => {
      expect(true).toBe(false);
    }).toFailTestWith(/expect.*[\s\S].*false.*[\s\S].*true/i);
  });

  test('toHaveBeenCalledWithArgumentsMatching exists and works', () => {
    const fn = jest.fn();
    fn('hello');
    expect(fn).toHaveBeenCalledWithArgumentsMatching(['hello']);
  });
});
