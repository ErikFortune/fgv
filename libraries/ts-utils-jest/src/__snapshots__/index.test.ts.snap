// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify matchers accessibility toFailTestAndMatchSnapshot exists and works: toFailTestAndMatchSnapshot 1`] = `
"expect(received).toBe(expected) // Object.is equality

Expected: false
Received: true"
`;

exports[`verify matchers accessibility toSucceedAndMatchSnapshot exists and works: toSucceedAndMatchSnapshot 1`] = `
Object {
  "child": Object {
    "values": Array [
      Object {
        "number": 1,
      },
      Object {
        "number": 2,
      },
    ],
  },
  "field": "field",
}
`;
